### PR TITLE
circle-ci 추가

### PR DESCRIPTION
**Pull request by Jununymous**
circle-ci 설정 추가.

-----------------------------------------------------------------------
`feature/travis-ci` branch merged into here ---------------------------
-----------------------------------------------------------------------
* Add travis yml

* 기존 gitlab.ci 요소들 포팅.

* Change file name

* Branch condition 수정.

* 조건을 type push 로 추가.

* 누락된 condition 수정.

* CI_COMMIT_SHA -> TRAVIS_PULL_REQUEST_SLUG

* 잘못넣은 변수 수정.

* deploy를 제외한 빌드는 push and PR 시에도 실행 되도록 수정.

* Deploy 를 job 에서 별도로 분리 진행중 and git flow 방식 수정.

* circle-ci yml setting ing.

* circle ci Aws-s3 orb 사용하도록 수정.

* Gitlab ci 내장변수 교체.

* 간소화 할 수 있는 부분 정리.

* Alias 정리

* travis 제거 및 circle ci config 수정.

* fix merge block

* Set default excuter

* workflow add

* flix workflow

* Fix indent

* 오타 수정.

* Fix indent

* branch condition fix and change workflow

* Fix indent at set jobs

* Set command for every run steps

* set working directory

* Add checkout

* set checkout to right place

* fix Cache rule

* 테스트 중에 배포되지 않도록 aws access key들 임시로 넣어둠.

* 리뷰 반영 및 수정.

* fix error for test variables

* add remarks and debug mode

* circleci/aws 에러 수정.

* run command fix

* aws arguments 정리.

* add reusable setting.

* block to commands

* add builder commands multiline acceptance

* add multi-line options

* change Command passing way

* Add approval and and more keys. Fix artifacts to workspace

* fix save and restore cache

* Rollback Environment variables

* fix save cache key

* Set save cache key to single line string

* change workspace target

* User environment variables

* Set workspace to root of working_directory

* fix workspace dir

* Deploy docker image to python

* Change variables HOME to WORKING_DIRECTORY

* add set ca-certificates commands

* Change commands to command key name

* set build filter to only develop and add deploy test

* Manual trigger testr

* Remove workspace temporaly

* Change options

* AWS REGION 변수값 환경변수에서 넣어주는 처리 추가.

* 개발환경 배포 시에 개발 빌드 하도록 플로우 수정.

* 불필요한 필터 조건 제거 및 배포 테스트 잡 제거.

* Deploy job 하나로 묶어서 condition으로 분기하도록 수정.

* 오탈자 수정 및 캐싱 대상 변경.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+          name: deploy-production
+          requires:
+            - wait-approval
+          prod: true
+      - sourcemaps:
+          requires:
+            - build-production
+      - serverless:
+          requires:
+            - build-production
```
Comment (tirr-c) [inline .circleci/config.yml]: `wait-approval`을 기다리지 않고 서버리스 배포가 일어나도 되나요?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+        type: boolean
+        default: false
+      prod:
+        type: boolean
+        default: false
+    executor: default_excuter
+    steps:
+      - when:
+          condition: <<parameters.dev>>
+          stpes:
```
Comment (tirr-c) [inline .circleci/config.yml]:
```suggestion
          steps:
```

File: .circleci/config.yml (hunk 1 above)
Comment (Jununymous) [reply .circleci/config.yml]: 기존에는 production 배포만 기다리는 것으로 하려고 했던 것 같은데, 아예 배포 자체를 다 묶어놓는게 일단 좋을 것 같아요.

File: .circleci/config.yml (hunk 2 above)
Comment (Jununymous) [reply .circleci/config.yml]: 안그래도 멘붕와서 헤메고 있었는데 이거 때문이었어요..

**Comment by Jununymous** [comment]
executer -> executor 로 수정했어요~

**Comment by Jununymous** [comment]
slack은 일단 배포 기능 먼저 머지하고 추후에 붙일게요 :)

**Comment by Jununymous** [comment]
@tirr-c @J132134 develop으로 머지하겠습니다~

**Pull request closed without merging**